components/base.templ

```diff
@@ -37,25 +41,55 @@ templ body() {
 	</body>
 }
 
-templ PageBase() {
+templ nav(signedIn bool) {
+	<nav>
+		if signedIn {
+			<a href={ templ.SafeURL("/") }>Home</a>
+			<a href={ templ.SafeURL("/my-templates") }>My Templates</a>
+			<a href={ templ.SafeURL("/profile") }>My Profile</a>
+			<a href={ templ.SafeURL("/logout") }>Logout</a>
+		} else {
+			<a href={ templ.SafeURL("/register") }>Register</a>
+			<a href={ templ.SafeURL("/login") }>Login</a>
+		}
+	</nav>
+}
+
+templ PageBase(flash []interface{}, signedIn bool, child templ.Component) {
 	@base() {
 		@head()
 		@body() {
-			<main>
-				{ children... }
+			@nav(signedIn)
+			<main class="width-wrap">
+				<h1>Chronologger</h1>
+				for _, flash := range flash {
+					@FlashMessage(flash)
+				}
+				@child
 			</main>
 		}
 	}
 }
 
-templ Home() {
+templ BaseBody(flash []interface{}, child templ.Component) {
 	@base() {
 		@head()
 		@body() {
 			<main class="width-wrap">
 				<h1>Chronologger</h1>
-				<a href={ templ.SafeURL("/templates") }>Our Templates</a>
+				for _, flash := range flash {
+					@FlashMessage(flash)
+				}
+				@child
 			</main>
 		}
 	}
 }
+
+templ FlashMessage(flash interface{}) {
+	<div>
+		A message
```

Comment:
```suggestion
```

One unanswered question is that I dont see in [the docs](https://pkg.go.dev/github.com/gorilla/sessions@v1.2.2#section-readme) a suggested way to pass in the type of flash this is, i.e. success, info etc. Perhaps we just store this in the same session, but another variable?

Reply:
See https://github.com/hay-i/chronologger/issues/44